beacon-chain/powchain/service.go

```diff
@@ -269,9 +269,14 @@ func (s *Service) ChainStartDeposits() []*ethpb.Deposit {
 }
 
 // ClearPreGenesisData clears out the stored chainstart deposits and beacon state.
-func (s *Service) ClearPreGenesisData() {
+func (s *Service) ClearPreGenesisData() error {
 	s.chainStartData.ChainstartDeposits = []*ethpb.Deposit{}
-	s.preGenesisState = &v1.BeaconState{}
+	newState, err := v1.InitializeFromProto(&ethpb.BeaconState{})
```

Comment:
how will this work if you initialize the native state via a zero proto object ?

Reply:
I don't see this as an issue. Currently the underlying `state` field is nil, so this seems better. The node runs fine. Is there anything in particular that you are worried about? This is the relevant code:
```
fieldCount := params.BeaconConfig().BeaconStateFieldCount
b := &BeaconState{
	state:                 st,
	dirtyFields:           make(map[types.FieldIndex]bool, fieldCount),
	dirtyIndices:          make(map[types.FieldIndex][]uint64, fieldCount),
	stateFieldLeaves:      make(map[types.FieldIndex]*fieldtrie.FieldTrie, fieldCount),
	sharedFieldReferences: make(map[types.FieldIndex]*stateutil.Reference, 10),
	rebuildTrie:           make(map[types.FieldIndex]bool, fieldCount),
	valMapHandler:         stateutil.NewValMapHandler(st.Validators),
}
```
As you can see `state` will become the empty proto. As for validators, I don't think having `st.Validators == nil` is a problem because we use `append` in the setter, and appending to a nil slice works fine in Go.
```
func (b *BeaconState) AppendValidator(val *ethpb.Validator) error {

	(...)

	b.state.Validators = append(vals, val)
```

Reply:
Linking the native state code here:
https://github.com/prysmaticlabs/prysm/blob/develop/beacon-chain/state/state-native/v1/state_trie.go#L40
If your provide an empty proto state struct, this method will try to access a non-existent field. However since we actually perform a range loop through those fields fortunately it doesn't panic from an OOB access. This would be an issue if it was using the traditional for loop. In that case, I don't see there being too much of an issue.

Reply:
Ah, thanks for clarifying. I will add a defensive check.